lib/huginn_bigcommerce_product_agent/bigcommerce_product_agent.rb

```diff
@@ -62,6 +63,10 @@ def validate_options
             unless options['mode'].present? && modes.include?(options['mode'])
                 errors.add(:base, "mode is a required field and must be one of: #{modes.join(', ')}")
             end
+
+            unless options['not_purchasable_format_list'].present?
+                errors.add(:base, 'not_purchasable_format_list is a required field')
+            end
```

Comment:
@Jhalter5Stones If the default options use an empty array, then this shouldn't be required.

If you have logic that expects an array later in the code, default it to an empty array there if it's not provided in the config